jena-arq/src/test/java/org/apache/jena/riot/writer/TestTurtleWriter.java

```diff
@@ -55,70 +60,110 @@ static void blankNodeLang(String testdata, RDFFormat lang) {
         Assert.assertTrue(m.size() > 0);
-        
+
         ByteArrayOutputStream output = new ByteArrayOutputStream();
         RDFDataMgr.write(output, m, Lang.TURTLE);
-        
+
         ByteArrayInputStream input = new ByteArrayInputStream(output.toByteArray());
         Model m2 = ModelFactory.createDefaultModel();
         RDFDataMgr.read(m2, input, Lang.TURTLE);
         Assert.assertTrue(m2.size() > 0);
-        
+
         Assert.assertTrue(m.isIsomorphicWith(m2));
     }
-    
+
+    // @base
     @Test
-    public void test_base() {
-        InputStream r = new ByteArrayInputStream(TestTurtleWriter.basetester.getBytes()) ;
-        String b = "http://example.org/" ;
-        Model m = ModelFactory.createDefaultModel() ;
-        m.read(r, b, "TTL") ;
+    public void test_base_1() {
+        // Default base style
+        String result = modelToString(baseTestData, RDFFormat.TURTLE_FLAT, null);
+        int count1 = StringUtils.countMatches(result, "@base");
+        Assert.assertEquals(1, count1);
+        int count2 = StringUtils.countMatches(result, "BASE");
+        Assert.assertEquals(0, count2);
+    }
+
+    @Test
+    public void test_base_2() {
+        Context cxt = RIOT.getContext().copy();
+        cxt.set(RIOT.symTurtleDirectiveStyle, DirectiveStyle.AT);
+        String result = modelToString(baseTestData, RDFFormat.TURTLE_BLOCKS, null);
+        int count1 = StringUtils.countMatches(result, "@base");
+        Assert.assertEquals(1, count1);
+        int count2 = StringUtils.countMatches(result, "BASE");
+        Assert.assertEquals(0, count2);
+    }
+
+
+    // BASE
+    @Test
+    public void test_base_3() {
+        Context cxt = RIOT.getContext().copy();
+        cxt.set(RIOT.symTurtleDirectiveStyle, DirectiveStyle.SPARQL);
+        String result = modelToString(baseTestData, RDFFormat.TURTLE_FLAT, cxt);
+        int count1 = StringUtils.countMatches(result, "BASE");
+        Assert.assertEquals(1, count1);
+        int count2 = StringUtils.countMatches(result, "@base");
+        Assert.assertEquals(0, count2);
+    }
+
+    @Test
+    public void test_base_4() {
+        Context cxt = RIOT.getContext().copy();
+        cxt.set(RIOT.symTurtleDirectiveStyle, DirectiveStyle.SPARQL);
+        String result = modelToString(baseTestData, RDFFormat.TURTLE_BLOCKS, cxt);
+        int count1 = StringUtils.countMatches(result, "BASE");
+        Assert.assertEquals(1, count1);
+        int count2 = StringUtils.countMatches(result, "@base");
+        Assert.assertEquals(0, count2);
+    }
 
-        OutputStream o = new ByteArrayOutputStream() ;
-        RDFWriter.create().source(m).format(RDFFormat.TURTLE_BLOCKS).base(b).output(o) ;
-        String k = o.toString() ;
-        Assert.assertTrue(k.contains("@base")) ;
+    private String modelToString(Model model, RDFFormat format, Context context) {
+        ByteArrayOutputStream o = new ByteArrayOutputStream();
```

Comment:
Should we use `try-with-resources` here to close it later? Test only so I think it may not matter much.

Reply:
It's a ByteArrayOutputStream so no external resources. 
Change made.

(Hmm - now we need to handle IOException!)